src/schedule/dma_multi_chan_domain.c

```diff
@@ -52,10 +52,25 @@ const struct ll_schedule_domain_ops dma_multi_chan_domain_ops;
 static void dma_multi_chan_domain_irq_handler(void *data)
 {
 	struct dma_domain_data *domain_data = data;
+	struct ll_schedule_data *sch = domain_data->arg;
+	struct dma_domain *dma_domain = ll_sch_domain_get_pdata(sch->domain);
+	struct dma *dmas = dma_domain->dma_array;
+	int i;
+	int j;
 
-	/* just call registered handler */
+	/* registered handler */
 	domain_data->handler(domain_data->arg);
 
+	/* clear inerrupts */
+	for (i = 0; i < dma_domain->num_dma; ++i) {
+		for (j = 0; j < dmas[i].plat_data.channels; ++j) {
+			if (dma_interrupt(&dmas[i].chan[j], DMA_IRQ_STATUS_GET)) {
+				dma_interrupt(&dmas[i].chan[j], DMA_IRQ_CLEAR);
+				interrupt_clear_mask(dma_domain->data[i][j].irq, BIT(j));
```

Comment:
> this looks suspicious from a superficial reading: _usually_ it isn't a good idea to clear IRQ sources other than the one, that triggered the interrupt. I'd at least want a comment here, explaining why this is the correct thing to do here. Even for DMA channels, owned by other cores. And maybe this would deserve a separate patch if possible

I didn't meant to change this, but during I changing the ll_scheduler logic that I realize the _is_pending() callback should only read back status and better to clear interrupts after the whole handler is finished. But yes, maybe let me keep it as first, and simplify the _is_pending() and interrupt clear next for easy reviewing.